File: microservices/catalog-service/src/main/java/xyz/tonyhe/catalogservice/domain/Product.java
package xyz.tonyhe.catalogservice.domain;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Document
@Data
@Builder
public class Product {

    @Id
    private long id;

    @NotNull
    private String name;

    private String description;

    private BigDecimal price;

    private int totalInventory;


}

File: ui/istore-angular-ui/src/app/models/Product.ts
export class Product {
    productId:number;
    name:string;
    description:string;
    price:number;
    quantity:number;

    constructor(
        productId: number, 
        name: string, 
        description: string,
        price: number,
        quantity: number) {
    this.productId = productId;
    this.name = name;
    this.description = description;
    this.price = price;
    this.quantity = quantity;
    }
}